packages/legend-query-builder/src/stores/watermark/QueryBuilderWatermarkValueSpecificationBuilder.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright (c) 2020-present, Goldman Sachs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  SimpleFunctionExpression,
+  extractElementNameFromPath,
+  type VariableExpression,
+} from '@finos/legend-graph';
+import { QUERY_BUILDER_SUPPORTED_FUNCTIONS } from '../../graphManager/QueryBuilderSupportedFunctions.js';
+import type { QueryBuilderWatermarkState } from './QueryBuilderWatermarkState.js';
+
+export const buildWatermarkExpression = (
+  watermarkState: QueryBuilderWatermarkState,
+  getAllFunc: SimpleFunctionExpression,
```

Comment:
is filter before forwaterMark or after? see one of the asserstions in QueryBuilderWatermarkStateBuilder.ts